docs/pull_request.md

```diff
@@ -0,0 +1,208 @@
+
+Big PR can take time, and there is a risk of future merge conflict.
+
+Feature flag can be used to avoid half implemented feature to be available in the application.
+
+That said, splitting into several PRs should not have the side effect to have more review to do, for instance if some code is added, then finally removed.
+
+##### Avoid fixing other unrelated issue in a big PR
+
+Each PR should focus on a single task. If other issues may be fixed when working in the area of it, it's preferable to open a dedicated PR.
+
+It will have the advantage to be reviewed and merged faster, and not interfere with the main PR.
+
+It's also applicable for code rework, or code formatting. Sometimes, it is more efficient to extract that work to a dedicated PR, and rebase your branch once this "rework" PR has been merged.
+
+#### Bots
+
+Some bots can create PR, but they still have to be reviewed by the team
+
+##### Dependabot
+
+Dependabot is a tool which maintain all our external dependencies up to date. A dedicated PR is created for each new available release for one of our external dependency.Dependabot
+
+To review such PR, you have to
+ - **IMPORTANT** check the diff files (as always).
+ - Check the release note. Some existing bugs in Element project may be fixed by the upgrade
+ - Make sure that the CI is happy
+ - If the code does not compile (API break for instance), you have to checkout the branch and push new commits
+ - Do some smoke test, depending of the library which has been upgraded
+
+For some reason dependabot sometimes does not upgrade some dependencies. In this case, and when detected, the upgrade has to be done manually.
+
+##### Gradle wrapper
+
+`Update Gradle Wrapper` is a tool which can create PR to upgrade our gradle.properties file.
+Review such PR is the same recipe than for PR from Dependabot
+
+##### Sync analytics plan
+
+This tools imports any update in the analytics plan. See instruction in the PR itself to handle it.
+More info can be found in the file [analytics.md]
+
+## Reviewing PR
+
+### Who can review pull requests?
+
+As an open source project, every one can review each others PR. Of course an approval from internal developer is mandatory for a PR to be merged.
+But comment in PR from the community are always appreciated!
+
+### What to have in mind when reviewing a PR
+
+1. User experience: is the UX and UI correct? You will probably be the second person to test the new thing, the first one is the developer.
+2. Developer experience: does the code look nice and decoupled? No big functions, new classes added to the right module, etc.
+3. Code maintenance. A bit similar to point 2. Tricky code must be documented for instance
+4. Fork consideration. Will configuration of forks be easy? Some documentation may help in some cases.
+5. We are building long term products. "Quick and dirty" code must be avoided.
```

Comment:
do you think it's worth including some examples? I've noticed a trend in community PRs commenting out existing logic rather than deleting the no longer used code   

this could probably be a separate doc for coding guidelines :sweat_smile: